lib/counter_culture.rb

```diff
@@ -15,6 +15,38 @@ def self.config
     yield(self) if block_given?
     self
   end
+
+  def self.aggregate_counter_updates
+    return unless block_given?
+
+    Thread.current[:aggregate_counter_updates] = true
+    Thread.current[:aggregated_updates] = {}
+    Thread.current[:primary_key_map] = {}
+
+    result = yield
+
+    # aggregate the updates for each target record and execute SQL queries
+    Thread.current[:aggregated_updates].each do |klass, attrs|
+      attrs.each do |rec_id, updates|
+        update_snippets = updates.map do |operation, value|
+          value = value.call if value.is_a?(Proc)
+          %Q{#{operation} #{value.is_a?(String) ? "'#{value}'" : value}} unless value == 0
+        end.compact
```

Comment:
@magnusvk Is there a reason this gem still supports Ruby 2.6 (which is EOL since 31 Mar 2022)?
Ruby 2.7 added filter_map, so if the gem will drop support for 2.6, this could be simplified.
```suggestion
        update_snippets = updates.filter_map do |operation, value|
          value = value.call if value.is_a?(Proc)
          %Q{#{operation} #{value.is_a?(String) ? "'#{value}'" : value}} unless value == 0
        end
```

Reply:
No reason, really, I tend to drop support when supporting the old version starts creating headaches, rather than when a version goes EOL. Aka I would be very open to dropping support for 2.6. That being said, I'm not sure I prefer `filter_map` over `map.compact`—I find the former to be just as readable, if not more so?

Reply:
I have been using filter_map for a while now, so for me it feels naturally to use it. I dislike chaining methods to end, but that is just a personal preference. 
I was curious if there was a reason to keep supporting older Ruby versions, but it is totally fine for me to keep it as is and decide on dropping 2.6 at a later stage when it brings more benefit.